### DISCUSSION TITLE
way to upload and run code on hub without code.pybricks.com

### DISCUSSION THREAD

**Discussion by Thedmcoder** [pybricks/support#1724] [Q&A] [answered]
So I'm new to pybricks, and I was just wondering if there was a way for me to run code on the hub from an application like pycharm instead of  putting the code onto code.pybricks.com

**Comment by dlech** [comment]
Yes, we have the `pybricksdev` command line tool for this. 

See:
* https://pybricks.com/project/pybricks-other-editors/
* https://github.com/pybricks/pybricksdev/tree/master#running-pybricks-micropython-programs.

**Reply by Thedmcoder** [reply]
oh okay thanks!

**Reply by Thedmcoder** [reply]
It looked like pybricksdev can only upload and execute a single program, but ultimately I need to upload multiple programs (for FLL competitions), as is described here: https://pybricks.com/project/spike-hub-menu/ - Is there a way to do that?  If all else fails, maybe I can use the pybricks UI for this step when the time comes, but I’d rather be able to do it all from PyCharm.  Thanks again!

**Reply by dlech** [reply, marked as answer]
`pybricksdev` works the same way. You just specify your main entry point program with the menu and it will include all imported modules (missions).

**Reply by FLL-Team-24277** [reply]
We used pybricks in 2023 Masterpiece and are getting set up for it again this year. There are several other teams using pybricks as well. https://github.com/FLL-Team-24277
